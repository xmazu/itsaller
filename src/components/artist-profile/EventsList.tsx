import React, { SFC } from 'react';

import './artist-profile.scss';
import { EventEntity } from '../../types';
import ArtistEvent from './ArtistEvent';

export interface EventsListProps {
  events: EventEntity[];
}

export interface EventsListState {
  openedId: string | null;
}

export default class EventsList extends React.Component<
  EventsListProps,
  EventsListState
> {
  constructor(props: EventsListProps) {
    super(props);

    this.state = {
      openedId: null
    };
  }

  onToggleEvent = (event: EventEntity) => () =>
    this.setState(prevState => ({
      openedId: prevState.openedId === event.id ? null : event.id
    }));

  renderEventItem = (event: EventEntity) => (
    <ArtistEvent
      key={event.id}
      event={event}
      opened={this.state.openedId === event.id}
      onToggle={this.onToggleEvent(event)}
    />
  );

  render() {
    const { events } = this.props;

    return (
      <div className="eventsList">
        {events.length > 0 ? (
          events.map(this.renderEventItem)
        ) : (
          <span>No events</span>
        )}
      </div>
    );
  }
}
