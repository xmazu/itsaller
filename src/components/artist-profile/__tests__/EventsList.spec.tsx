import React from 'react';
import { shallow } from 'enzyme';

import EventsList from '../EventsList';

describe('<EventsList />', () => {
  it('renders correctly the list', () => {
    const mockedEvent = {
      id: 'my.id',
      date: '2018-12-30T19:00:48',
      venue: 'Tauron Arena',
      city: 'Krakow',
      country: 'Poland',
      lat: 0,
      long: 0
    };
    const component = shallow(<EventsList events={[mockedEvent]} />);
    expect(component).toMatchSnapshot();
  });

  it('renders message for empty list', () => {
    const component = shallow(<EventsList events={[]} />);
    expect(component.text()).toEqual('No events found.');
  });
});
