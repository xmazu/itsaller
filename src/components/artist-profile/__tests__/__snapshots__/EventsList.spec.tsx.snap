// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventsList /> renders correctly the list 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventsList
    events={
      Array [
        Object {
          "city": "Krakow",
          "country": "Poland",
          "date": "2018-12-30T19:00:48",
          "id": "my.id",
          "lat": 0,
          "long": 0,
          "venue": "Tauron Arena",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ArtistEvent
          event={
            Object {
              "city": "Krakow",
              "country": "Poland",
              "date": "2018-12-30T19:00:48",
              "id": "my.id",
              "lat": 0,
              "long": 0,
              "venue": "Tauron Arena",
            }
          }
          onToggle={[Function]}
          opened={false}
        />,
      ],
      "className": "eventsList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "my.id",
        "nodeType": "function",
        "props": Object {
          "event": Object {
            "city": "Krakow",
            "country": "Poland",
            "date": "2018-12-30T19:00:48",
            "id": "my.id",
            "lat": 0,
            "long": 0,
            "venue": "Tauron Arena",
          },
          "onToggle": [Function],
          "opened": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ArtistEvent
            event={
              Object {
                "city": "Krakow",
                "country": "Poland",
                "date": "2018-12-30T19:00:48",
                "id": "my.id",
                "lat": 0,
                "long": 0,
                "venue": "Tauron Arena",
              }
            }
            onToggle={[Function]}
            opened={false}
          />,
        ],
        "className": "eventsList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "my.id",
          "nodeType": "function",
          "props": Object {
            "event": Object {
              "city": "Krakow",
              "country": "Poland",
              "date": "2018-12-30T19:00:48",
              "id": "my.id",
              "lat": 0,
              "long": 0,
              "venue": "Tauron Arena",
            },
            "onToggle": [Function],
            "opened": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
